### PR TITLE
Implement TTS

### PR DESCRIPTION
**Pull request by dvonthenen**
updayte

## Proposed changes

Addresses issue: https://github.com/deepgram/deepgram-go-sdk/issues/190

This PR implements TTS. This enables the ability to:
- use in-memory data via `ToStream()`
- a user-defined file via `ToFile()`
- save to a defined path `ToSave()`

Contains examples for all of the methods above in the `examples/speak` folder.

Other smaller changes:
- Fixed a bunch of incorrect comments
- Removed some dead code `Do()` in a number of the clients.

This will be released at `v1.2`.

## Types of changes

What types of changes does your code introduce to the community Go SDK?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update or tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I have lint'ed all of my code using repo standards
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

NA

### CODE REVIEW

File: pkg/client/interfaces/types-analyze.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ package interfaces
 AnalyzeOptions contain all of the knobs and dials to control a read transcription
 from the Deepgram API
 
-Please see the documentation for live/streaming for more details:
+Please see the live/streaming documentation for more details:
```
Comment (SandraRodgers) [inline pkg/client/interfaces/types-analyze.go:11]: Should this say live/streaming? Probably should say Text Intelligence documentation

File: pkg/client/interfaces/types-prerecorded.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ package interfaces
 PreRecordedTranscriptionOptions contain all of the knobs and dials to control a Prerecorded transcription
 from the Deepgram API
 
-Please see the documentation for live/streaming for more details:
+Please see the live/streaming documentation for more details:
```
Comment (SandraRodgers) [inline pkg/client/interfaces/types-prerecorded.go:11]: Should say "see the prerecorded documentation"

File: pkg/client/interfaces/types-analyze.go (hunk 1 above)
Comment (dvonthenen) [reply pkg/client/interfaces/types-analyze.go:11]: ooops. good catch! copy and paste is not your friend in many cases (like this)

File: pkg/client/interfaces/types-prerecorded.go (hunk 2 above)
Comment (dvonthenen) [reply pkg/client/interfaces/types-prerecorded.go:11]: will update! another silly copy and paste error.

**Comment by dvonthenen** [comment]
Thanks for the feedback @SandraRodgers . Updated the PR to reflect those changes.

**Review by SandraRodgers — APPROVED** [review]

**Pull request merged**